service/frontend/workflow_handler.go

```diff
@@ -4183,31 +4183,42 @@ func (wh *WorkflowHandler) getHistory(
 	branchToken []byte,
 ) (*historypb.History, []byte, error) {
 
+	// readFullPageEvents reads a full page of history events from history service. Due to storage format of V2 History
+	// it is not guaranteed that pageSize amount of data is returned. Function returns the list of history events, the
+	// size of data read, the next page token, and an error if present.
+	readFullPageEvents := func(
+		req *historyservice.ReadHistoryBranchRequest,
+	) ([]*historypb.HistoryEvent, int, []byte, error) {
```

Comment:
I'm with you. But I think fixing anything beyond the goals of this PR should be separate.

The reason why I moved it from `history_manager_util.go` to here as a lambda is because our linter recently started to not allow returning more than 3 values per function. As I mentioned in my other response, I'm deliberately porting the code over as close to the existing way as possible.